plugins/inputs/amd_rocm_smi/amd_rocm_smi.go

```diff
@@ -1,294 +1,141 @@
 package amd_rocm_smi
 
 import (
-	"encoding/json"
+	_ "embed" // Required for embedding the parser config file
 	"fmt"
-	"os"
-	"os/exec"
 	"strconv"
 	"strings"
 	"time"
 
+	"github.com/influxdata/toml"
+
 	"github.com/influxdata/telegraf"
 	"github.com/influxdata/telegraf/config"
-	"github.com/influxdata/telegraf/internal"
+	"github.com/influxdata/telegraf/plugins/common/transport"
 	"github.com/influxdata/telegraf/plugins/inputs"
-)
-
-const measurement = "amd_rocm_smi"
+	"github.com/influxdata/telegraf/plugins/parsers"
 
-type ROCmSMI struct {
-	BinPath string
-	Timeout config.Duration
-}
-
-// Description returns the description of the ROCmSMI plugin
-func (rsmi *ROCmSMI) Description() string {
-	return "Query statistics from AMD Graphics cards using rocm-smi binary"
-}
-
-var ROCmSMIConfig = `
-## Optional: path to rocm-smi binary, defaults to $PATH via exec.LookPath
-# bin_path = "/opt/rocm/bin/rocm-smi"
+	generic "github.com/influxdata/telegraf/plugins/common/receive_parse"
+)
 
-## Optional: timeout for GPU polling
-# timeout = "5s"
-`
+//go:embed amd_rocm_smi_parser.conf
+var cfgfile []byte
 
-// SampleConfig returns the sample configuration for the ROCmSMI plugin
-func (rsmi *ROCmSMI) SampleConfig() string {
-	return ROCmSMIConfig
-}
-
-// Gather implements the telegraf interface
-func (rsmi *ROCmSMI) Gather(acc telegraf.Accumulator) error {
-	if _, err := os.Stat(rsmi.BinPath); os.IsNotExist(err) {
-		return fmt.Errorf("rocm-smi binary not found in path %s, cannot query GPUs statistics", rsmi.BinPath)
+func NewAMDSMI() *generic.ReceiveAndParse {
+	var cfg parsers.Config
+	if err := toml.Unmarshal(cfgfile, &cfg); err != nil {
+		panic(fmt.Errorf("cannot unmarshal 'amd_rocm_smi_parser.conf': %v", err))
 	}
-
```

Comment:
Let's do revert those commits. It's clearer to put them in static structs in the code, and has the benefit of failing at compile time instead of runtime if there's a typo.